### ISSUE TITLE
Scheduling: reserve CPU cores for upcoming GPU tasks

### ISSUE THREAD

**Issue by mtitov** [radical-cybertools/radical.pilot#3280]
Reserve a CPU core per GPU within a node, if there are enough resources on other available compute nodes to launch a new coming task

**Comment by andre-merzky** [comment]
We should  discuss what the actual policy is.  Like, we don't want to stop full-node CPU-only tasks either, right?  So should we make this a config option (`cpu_task_max_cores`), or should we mark tasks to lower priority on CPU-only ones, or should we decide dynamically - and how?

**Comment by mtitov** [comment]
@andre-merzky should we park this ticket?

**Comment by andre-merzky** [comment]
Yes, I think parking until a good use case really requires that functionality makes sense.  Otherwise it is hard to decide what reservation policy to use anyway...
